protocol/test/forge/v2/strategies/RamosStrategy.t.sol

```diff
@@ -0,0 +1,458 @@
+        bptTotalSupply = IBPT(address(bptToken)).getActualSupply();
+
+        if (bptTotalSupply > 0) {
+            bptBalance = amoStaking.stakedBalance() + bptToken.balanceOf(address(amoStaking));
+
+            (uint256 templeBalanceInLP, uint256 stableBalanceInLP) = PoolHelper(POOL_HELPER_ADDRESS).getTempleStableBalances();
+            templeBalance = templeBalanceInLP * bptBalance / bptTotalSupply;
+            stableBalance = stableBalanceInLP * bptBalance / bptTotalSupply;
+        }
+    }
+
+    function test_borrowAndAddLiquidity() public {
+        uint256 amount = 1e18;
+        uint256 slippageBps = 100;  // 1%
+        ITempleStrategy.AssetBalance[] memory assetBalances;
+
+        assertEq(dai.balanceOf(address(trv)), trvStartingBalance);
+
+        (uint256 debt, uint256 available, uint256 ceiling) = strategy.trvBorrowPosition();
+        assertEq(debt, 0);
+        assertEq(available, borrowCeiling);
+        assertEq(ceiling, borrowCeiling);
+
+        (, uint256 bptBalanceBefore, uint256 templeBalanceBefore, uint256 stableBalanceBefore) = getRamosBalances();
+        (assetBalances,) = strategy.latestAssetBalances();
+        uint256 daiBalanceBefore = assetBalances[0].balance;
+
+        (, uint256 bptOut,, AMO__IBalancerVault.JoinPoolRequest memory requestData) = PoolHelper(POOL_HELPER_ADDRESS).proportionalAddLiquidityQuote(amount, 5000, slippageBps);
+
+        vm.expectEmit();
+        emit BorrowAndAddLiquidity(amount);
+
+        vm.startPrank(executor);
+        strategy.borrowAndAddLiquidity(amount, requestData);
+        vm.stopPrank();
+
+        (assetBalances,) = strategy.latestAssetBalances();
+        uint256 daiBalanceAfter = assetBalances[0].balance;
+
+        assertEq(dai.balanceOf(address(strategy)), 0);
+        assertEq(dai.balanceOf(address(trv)), trvStartingBalance-amount);
+        // When adding liquidity into the weighted pool, BPTs more than the expected `bptOut' are minted as a protocol fee and transferred to the protocol fee collector address.
+        // https://docs.balancer.fi/concepts/pools/more/deployments.html
+        // https://etherscan.io/tx/0x1ac228b6582ae5774de4dd508de0ca21c4e0be31e1ae27428e50bcc43acabbe3
+        // Thus, the share is decreased and the token balances corresponding to the share are also decreased
+        // check the balance with delta. 1e18 = 100%
+        assertApproxEqRel(daiBalanceAfter-daiBalanceBefore, amount, 0.0001e18); // 0.001% 
+
+        assertEq(dUSD.balanceOf(address(strategy)), amount);
+        assertEq(dUSD.balanceOf(address(trv)), 0);
+
+        (debt, available, ceiling) = strategy.trvBorrowPosition();
+        assertEq(debt, amount);
+        assertEq(available, 0.01e18);
+        assertEq(ceiling, borrowCeiling);
+
+        (, uint256 bptBalanceAfter, uint256 templeBalanceAfter, uint256 stableBalanceAfter) = getRamosBalances();
```

Comment:
I think it would be worth changing the amount added to be a large amount of liquidity proportional to the pool size (ie 10 million worth) -- or add another test for this. 5000 wei is tiny proportionally

It's important that we've nailed this logic or we might affect the pool in a bad way which would cost $$

I was contemplating doing `BalancerQueries.querySwap()`, but if we add a big amount that would give a different rate because the price impact of the next swap would be much lower.
https://docs.balancer.fi/reference/contracts/query-functions.html#queryswap